### ISSUE TITLE
`silk-paper` release requirements

### ISSUE THREAD

**Issue by jakobkmar** [SilkMC/silk#57]
Since Paper removed the version strings from most packages, it is easier now that ever to fully implement `silk-paper`.
In the past year, I also removed all dependencies on `fabric-api`, so Silk is standalone in that regard, 

What still needs to be done:
- **all** Silk events also need to be called on paper
  - which should be done like this:
    1. listen to paper event
    2. call silk event inside there
    3. handle changed state correctly
  - I aim to write a simple compiler plugin (be it KSP or possibly something with K2) to check whether all events have been implemented, or fail / warn otherwise - but this is not a requirement for the first silk-paper release
- Silk sometimes uses accessors on Fabric to access private / protected Minecraft fields and classes, this probably needs a reflection replacement on paper

I won't be able to begin with this within the next 1 1/2 months, however if anyone wants to take the lead and get this done it would be greatly appreciated.

**Comment by DinoMarlir** [comment]
Silk-Paper will replace [KSpigot](https://github.com/jakobkmar/KSpigot), right?

**Comment by jakobkmar** [comment]
Yes, KSpigot can be continued by anyone who wants to maintain it, but I would definitely recommend to switch since the APIs in Silk are more refined.
